lib/db_connection/ownership/owner.ex

```diff
@@ -92,6 +77,24 @@ defmodule DBConnection.Ownership.Owner do
     {:noreply, state}
   end
 
+  def handle_call(:init, _from, state) do
+    %{pool: pool, pool_mod: pool_mod, pool_opts: pool_opts} = state
+
+    try do
+      pool_mod.checkout(pool, pool_opts)
+    catch
+      kind, reason ->
+        {:stop, {:shutdown, "no checkout"}, {kind, reason, System.stacktrace}, state}
```

Comment:
Lets reply with `:error` and re-raise the exception. Should not reraise an exception from another process as the stacktrace is impossible and it is confusing.